grammar/it/src/rules.rs

```diff
@@ -0,0 +1,513 @@
+                     latent: false,
+                     ..*a.value()
+                 })
+             });
+    Ok(())
+}
+
+pub fn rules_numbers(b: &mut RuleSetBuilder<Dimension>) -> RustlingResult<()> {
+    b.rule_1_terminal("number (0..19)",
+                      b.reg(r#"(zero|un[oa']?|due|tre|quattro|cinque|sei|sette|otto|nove|dieci|(?:undici|dodici|tredici|quattordici|quindici|sedici)|(?:dici(?:assette|otto|annove)))"#)?,
+                      |text_match| {
+                          let value = match text_match.group(1).as_ref() {
+                              "zero" => 0,
+                              "un" => 1,
+                              "un'" => 1,
+                              "uno" => 1,
+                              "una" => 1,
+                              "due" => 2,
+                              "tre" => 3,
+                              "quattro" => 4,
+                              "cinque" => 5,
+                              "sei" => 6,
+                              "sette" => 7,
+                              "otto" => 8,
+                              "nove" => 9,
+                              "dieci" => 10,
+                              "undici" => 11,
+                              "dodici" => 12,
+                              "tredici" => 13,
+                              "quattordici" => 14,
+                              "quindici" => 15,
+                              "sedici" => 16,
+                              "diciassette" => 17,
+                              "diciotto" => 18,
+                              "diciannove" => 19,
+                              _ => return Err(RuleError::Invalid.into()),
+                          };
+                          IntegerValue::new(value)
+                      }
+    );
+    b.rule_1_terminal("number (20..90)",
+                      b.reg(r#"(venti|trenta|(?:(?:quar|cinqu|sess|sett|ott|nov)anta))"#)?,
+                      |text_match| {
+                          let value = match text_match.group(1).as_ref() {
+                              "venti" => 20,
+                              "trenta" => 30,
+                              "quaranta" => 40,
+                              "cinquanta" => 50,
+                              "sessanta" => 60,
+                              "settanta" => 70,
+                              "ottanta" => 80,
+                              "novanta" => 90,
+                              _ => return Err(RuleError::Invalid.into()),
+                          };
+                          IntegerValue::new(value)
+                      });
+    b.rule_2("number (21..29 31..39 41..49 51..59 61..69 71..79 81..89 91..99)",
+             b.reg(r#"(venti?|trenta?|(?:(?:quar|cinqu|sess|sett|ott|nov)anta?))"#)?,
+             integer_check_by_range!(1, 9),
+             |text_match, b| {
+                 let value = match text_match.group(1).as_ref() {
+                     "venti" => 20,
```

Comment:
the rule `number (21..29 31..39 41..49 51..59 61..69 71..79 81..89 91..99)` can be replace by https://github.com/snipsco/rustling-ontology/blob/develop/grammar/en/src/rules.rs#L1690

However, if you want to reduce composition during parsing, your approach is better because rules are more precise.

Reply:
OK, will explore compose_numbers() for next version when we handle Italian more exhaustively.